src/main/src/main/java/org/geoserver/catalog/impl/LocalWorkspaceCatalog.java

```diff
@@ -160,11 +160,28 @@ public LayerGroupInfo getLayerGroup(String id) {
     @Override
     public LayerGroupInfo getLayerGroupByName(String name) {
         if (LocalWorkspace.get() != null) {
-            LayerGroupInfo layerGroup = super.getLayerGroupByName(LocalWorkspace.get(), name);
+            String wsName = LocalWorkspace.get().getName();
+
+            //prefix the unqualified name
+            if (name.contains(":")) {
+                //name already prefixed, ensure it is prefixed with the correct one
+                if (name.startsWith(wsName+":")) {
+                    //good to go, just pass call through
+                    LayerGroupInfo layerGroup = super.getLayerGroupByName(name);
+                    if (layerGroup != null) {
+                        return wrap(layerGroup);
+                    }
+                    // else fall back on unqualified lookup
+                }
+                else {
+                    //JD: perhaps strip of existing prefix?
```

Comment:
Formatting: should be "} else {" all on one line.

Also, who is JD? @jdeolive? Is this the restoration of removed code or code copied from elsewhere? I am not familiar enough with this to know whether it will cause problems. There have been a bunch of recent changes, but the last change to this method was from @groldan back in May 2012. @aaime, any opinion?

Reply:
It's an adaptation from the getLayerByName method: 
```
    @Override
    public LayerInfo getLayerByName(String name) {
        if (LocalWorkspace.get() != null) {
            String wsName = LocalWorkspace.get().getName();

            //prefix the unqualified name
            if (name.contains(":")) {
                //name already prefixed, ensure it is prefixed with the correct one
                if (name.startsWith(wsName+":")) {
                    //good to go, just pass call through
                    return wrap(super.getLayerByName(name));
                }
                else {
                    //JD: perhaps strip of existing prefix?
                }
            }

            //prefix it explicitly
            NamespaceInfo ns = super.getNamespaceByPrefix(LocalWorkspace.get().getName());
            LayerInfo layer = super.getLayerByName(new NameImpl(ns.getURI(), name));
            return wrap(layer);
        }
        return super.getLayerByName(name);
    }
```

The commit seems to be from jdeolive based on work from Juan Marin: https://github.com/sogelink/geoserver/commit/dbedaaea132ca88b2371107132ee7fd9271ca938 with commit message : "GEOS-5016, dropping layer name prefixes when accessing through local workspace / virtual service, original patch by Juan Marin"